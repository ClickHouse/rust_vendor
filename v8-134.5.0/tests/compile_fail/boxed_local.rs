// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut scope1 = v8::HandleScope::new(&mut isolate);

  let _boxed_local = {
    let mut scope2 = v8::HandleScope::new(&mut scope1);
    let mut scope3 = v8::HandleScope::new(&mut scope2);
    Box::new(v8::Integer::new(&mut scope3, 123))
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
