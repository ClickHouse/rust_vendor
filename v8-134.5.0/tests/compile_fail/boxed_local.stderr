error[E0597]: `scope2` does not live long enough
  --> tests/compile_fail/boxed_local.rs:9:43
   |
7  |   let _boxed_local = {
   |       ------------ borrow later stored here
8  |     let mut scope2 = v8::HandleScope::new(&mut scope1);
   |         ---------- binding `scope2` declared here
9  |     let mut scope3 = v8::HandleScope::new(&mut scope2);
   |                                           ^^^^^^^^^^^ borrowed value does not live long enough
10 |     Box::new(v8::Integer::new(&mut scope3, 123))
11 |   };
   |   - `scope2` dropped here while still borrowed
